drivers/soundwire/cadence_master.h

```diff
@@ -92,6 +93,7 @@ struct sdw_cdns_dai_runtime {
 	struct snd_pcm_hw_params *hw_params;
 	bool suspended;
 	bool paused;
+	int direction;
```

Comment:
@plbossart just like the snd_soc_dai has plaback_dma_data/capture_dma_data, should there be playback_private/capture_private as well?

Reply:
I thought about it, but this can be handled internally I think. Not all DAIs support two directions, SoundWire is typically single direction fixed in hardware, and when the port direction can be configured it's still not full-duplex.

So I am erring on the side of keeping things simple with a single pointer that doesn't assume anything.

in the specific case of this 'direction' field, it's only needed because when we suspend paused stream we do this at the component suspend level, so we don't have the substream->stream direction information. This is specific to SoundWire and how we deal with suspend of paused stream, it's not a generic solution - if there was a generic solution should be in the sound/core/ directory not SOF or SoundWire-specific.